twake/frontend/src/app/scenes/Client/ChannelsBar/ChannelWorkspaceEditor.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { FC, useState } from 'react';
+import Languages from 'services/languages/languages.js';
+import Button from 'components/Buttons/Button.js';
+import ChannelTemplateEditor from 'app/scenes/Client/ChannelsBar/ChannelTemplateEditor';
+import MediumPopupComponent from 'services/mediumPopupManager/mediumPopupManager.js';
+import { ObjectModal, ObjectModalTitle } from 'components/ObjectModal/ObjectModal.js';
+import Collections from 'app/services/CollectionsReact/Collections';
+import { ChannelType, ChannelResource } from 'app/models/Channel';
+
+type Props = {
+  title: string;
+  channel?: ChannelType;
+};
+
+const ChannelWorkspaceEditor: FC<Props> = ({ title, channel }) => {
+  const [newChannel, setNewChannel] = useState<ChannelType>({
+    name: '',
+    icon: '',
+    company_id: '',
+    workspace_id: '',
+    visibility: 'private',
+  });
+
+  const collectionPath: string = `/companies/${newChannel.company_id}/workspaces/${newChannel.workspace_id}/channels/`;
+  const ChannelsCollections = Collections.get(collectionPath);
+
+  const onChange = (channelEntries: ChannelType): void => {
+    return setNewChannel(Object.assign({}, channelEntries));
```

Comment:
We don't need to use a state for this, do we show the channel components in this component ? No we don't.
So you can use a simple variable to store the template object (all in one variable by the way).

Reply:
fixed :+1: